pkg/apis/triggers/v1alpha1/interceptor_type_defaults.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+)
+
+// SetDefaults sets the defaults on the object.
+func (it *InterceptorType) SetDefaults(ctx context.Context) {
+	if IsUpgradeViaDefaulting(ctx) {
+		if svc := it.Spec.ClientConfig.Service; svc != nil {
+			if svc.Port == 0 {
+				svc.Port = 80 // TODO: Should we use a pointer type instead of 0?
```

Comment:
🤷  0 isn't a valid port, so it probably doesn't matter.

Reply:
I'll switch to a pointer type to make this clearer.